src/store/transactions/force-exit/force-exit.thunks.ts

```diff
@@ -0,0 +1,108 @@
+import theme from "src/styles/theme";
+// domain
+import { Account, FiatExchangeRates, PoolTransaction } from "src/domain/hermez";
+import { createAccount } from "src/utils/accounts";
+
+/**
+ * Fetches the accounts to use in the transaction in the rollup api.
+ */
+function fetchAccounts(
+  fromItem: number | undefined,
+  poolTransactions: PoolTransaction[],
+  fiatExchangeRates: FiatExchangeRates,
+  preferredCurrency: string
+): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, tokensPriceTask },
+    } = getState();
+
+    if (wallet !== undefined) {
+      dispatch(forceExitActions.loadAccounts());
+
+      return CoordinatorAPI.getAccounts(wallet.publicKeyBase64, undefined, fromItem)
+        .then((res) => {
+          const accounts = res.accounts.map((account) =>
+            createAccount(
+              account,
+              poolTransactions,
+              undefined,
+              tokensPriceTask,
+              fiatExchangeRates,
+              preferredCurrency
+            )
+          );
+
+          return { ...res, accounts };
+        })
+        .then((res) => dispatch(forceExitActions.loadAccountsSuccess(res)))
+        .catch((err) => dispatch(forceExitActions.loadAccountsFailure(err)));
+    }
+  };
+}
+
+/**
+ * Fetches the transactions which are in the transactions pool
+ */
+function fetchPoolTransactions(): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    dispatch(forceExitActions.loadPoolTransactions());
+
+    const {
+      global: { wallet },
+    } = getState();
+
+    if (wallet !== undefined) {
+      getPoolTransactions(undefined, wallet.publicKeyCompressedHex)
+        .then((transactions) =>
+          dispatch(forceExitActions.loadPoolTransactionsSuccess(transactions))
+        )
+        .catch((err) => dispatch(forceExitActions.loadPoolTransactionsFailure(err)));
+    }
+  };
+}
+
+function forceExit(amount: BigNumber, account: Account) {
+  return (dispatch: AppDispatch, getState: () => AppState): void => {
+    const {
+      global: { signer },
+    } = getState();
+
+    dispatch(forceExitActions.startTransactionApproval());
+
+    if (signer) {
+      Tx.forceExit(
+        HermezCompressedAmount.compressAmount(amount.toString()),
+        account.accountIndex,
+        account.token,
+        signer
+      )
+        .then(() => handleTransactionSuccess(dispatch))
+        .catch((error) => {
+          console.error(error);
+          dispatch(forceExitActions.stopTransactionApproval());
+          handleTransactionFailure(dispatch, error);
+        });
+    }
+  };
+}
+
+function handleTransactionSuccess(dispatch: AppDispatch) {
+  dispatch(openSnackbar("Transaction submitted"));
+  dispatch(push("/"));
+}
+
+function handleTransactionFailure(dispatch: AppDispatch, error: Error | string) {
+  const errorMsg = error instanceof Error ? error.message : error;
```

Comment:
When we cancel the TX from Metamask, `error` is not an instance of `Error`. Metamask sends an object with a field `message` which is not an `Error`.

This is a mistake we have been spreading and that I've solved in the PR [#744](https://github.com/hermeznetwork/wallet-ui/pull/744). We should somehow include this case when both branches are merged.

You can easily see the problem in your branch by rejecting a force-exit  from Metamask and looking at the error:

![image](https://user-images.githubusercontent.com/15896177/141766774-1c6702b0-770b-4c55-937e-ca69824c7251.png)

Reply:
One option is for you to approve my PR first so I can merge it and then you can use the helper in this PR.

Reply:
Yes, we can merge your PR later and then I'll bring the changes to this branch.